host/sgx/loadelf.c

```diff
@@ -834,6 +1025,161 @@ static oe_result_t _sgx_update_enclave_properties(
     return result;
 }
 
+static oe_result_t _find_module_image(
+    const char* enclave_path,
+    oe_enclave_image_t* image,
+    char** module_path)
+{
+    oe_result_t result = OE_OK;
+    elf64_dyn_t* section_data = NULL;
+    size_t section_size = 0;
+    uint64_t number_of_entries;
+    elf64_addr_t strtab_offset = 0;
+    elf64_xword_t needed_offset = 0;
+    char* module_name = NULL;
+    char* path_name = NULL;
+    size_t path_size = 0;
+    const char* n = NULL;
+    const char* p = NULL;
+
+    assert(enclave_path);
+    assert(image);
+
+    if (module_path)
+        *module_path = NULL;
+
+    if (elf64_find_section(
+            &image->elf.elf,
+            ".dynamic",
+            (uint8_t**)&section_data,
+            &section_size) != 0)
+        goto done;
+
+    if (!section_data || !section_size)
+        OE_RAISE(OE_UNEXPECTED);
```

Comment:
Here an elsewhere with ELF parsing errors (e.g. lines 1082, 1087), consider `OE_INVALID_IMAGE` instead of the more generic `OE_UNEXPECTED`.

Reply:
Updated.